spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,199 @@
+import com.google.common.collect.Lists;
+import joptsimple.internal.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Spark plugin Utility class. Contains common code to be used in trainer and predictors.
+ */
+final class SparkUtils {
+
+  private SparkUtils() {
+  }
+
+  /**
+   * Validate the config parameters for the spark sink and spark compute classes.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the prediction values.
+   */
+  protected static void validateConfigParameters(Schema inputSchema, @Nullable String featuresToInclude,
+                                                 @Nullable String featuresToExclude, String predictionField,
+                                                 @Nullable String cardinalityMapping) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one.");
+    }
+    Map<String, Integer> fields = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, predictionField);
+    for (String field : fields.keySet()) {
+      Schema.Field inputField = inputSchema.getField(field);
+      Schema schema = inputField.getSchema();
+      Schema.Type features = schema.isNullableSimple() ? schema.getNonNullable().getType() : schema.getType();
+      if (!(features.equals(Schema.Type.INT) || features.equals(Schema.Type.LONG) ||
+        features.equals(Schema.Type.FLOAT) || features.equals(Schema.Type.DOUBLE))) {
+        throw new IllegalArgumentException(String.format("Field to classify must be continuous : int, double, " +
+                                                           "float, long but was %s.", features));
+      }
+    }
+    getCategoricalFeatureInfo(inputSchema, featuresToInclude, featuresToExclude, predictionField, cardinalityMapping);
+  }
+
+  /**
+   * Get the feature list of the features that have to be used for training/prediction depending on the
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the prediction values.
+   * @return feature list to be used for training/prediction.
+   */
+  protected static Map<String, Integer> getFeatureList(Schema inputSchema, @Nullable String featuresToInclude,
```

Comment:
This is intended to be used in the plugin transform and run methods.
Hence, keeping this protected